Mage.Sets/src/mage/cards/c/CaseOfTheGorgonsKiss.java

```diff
@@ -0,0 +1,149 @@
+
+    static {
+        filter.add(WasDealtDamageThisTurnPredicate.instance);
+    }
+
+    public CaseOfTheGorgonsKiss(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ENCHANTMENT}, "{B}");
+
+        this.subtype.add(SubType.CASE);
+
+        // When this Case enters the battlefield, destroy up to one target creature that was dealt damage this turn.
+        Ability initialAbility = new EntersBattlefieldTriggeredAbility(new DestroyTargetEffect())
+                .setTriggerPhrase("When this Case enters the battlefield, ");
+        initialAbility.addTarget(new TargetPermanent(0, 1, filter));
+        // To solve -- Three or more creature cards were put into graveyards from anywhere this turn.
+        Condition toSolveCondition = new CaseOfTheGorgonsKissCondition();
+        // Solved -- This Case is a 4/4 Gorgon creature with deathtouch and lifelink in addition to its other types.
+        Ability solvedAbility = new SimpleStaticAbility(new ConditionalContinuousEffect(
+                new BecomesCreatureSourceEffect(new CaseOfTheGorgonsKissToken(),
+                        CardType.ENCHANTMENT, Duration.WhileOnBattlefield),
+                SolvedSourceCondition.SOLVED, "")
+                .setText("This Case is a 4/4 Gorgon creature with deathtouch and lifelink in addition to its other types."));
+
+        this.addAbility(new CaseAbility(initialAbility, toSolveCondition, solvedAbility)
+                .addHint(new CaseOfTheGorgonsKissHint(toSolveCondition)),
+                new CardsPutIntoGraveyardWatcher());
+    }
+
+    private CaseOfTheGorgonsKiss(final CaseOfTheGorgonsKiss card) {
+        super(card);
+    }
+
+    @Override
+    public CaseOfTheGorgonsKiss copy() {
+        return new CaseOfTheGorgonsKiss(this);
+    }
+}
+
+class CaseOfTheGorgonsKissCondition implements Condition {
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        CardsPutIntoGraveyardWatcher watcher = game.getState().getWatcher(CardsPutIntoGraveyardWatcher.class);
+        return watcher != null &&
+                watcher.getCardsPutIntoGraveyardFromAnywhere(game)
+                        .stream()
+                        .filter(MageObject::isCreature)
+                        .count() >= 3;
+    }
+
+    @Override
+    public String toString() {
+        return "Three or more creature cards were put into graveyards from anywhere this turn";
+    }
+}
+
+class CaseOfTheGorgonsKissHint extends CaseSolvedHint {
+
+    CaseOfTheGorgonsKissHint(Condition condition) {
+        super(condition);
+    }
+
+    private CaseOfTheGorgonsKissHint(final CaseOfTheGorgonsKissHint hint) {
+        super(hint);
+    }
+
+    @Override
+    public CaseOfTheGorgonsKissHint copy() {
+        return new CaseOfTheGorgonsKissHint(this);
+    }
+
+    @Override
+    public String getConditionText(Game game, Ability ability) {
+        int creatures = (int)game.getState()
```

Comment:
(this cast looked concerning for a moment but then I realized it's just that `count()` returns a `long`)